core/commands/resolve.go

```diff
@@ -62,27 +70,19 @@ Resolve te value of another name:
 			}
 		}
 
-		var name string
-
-		if len(req.Arguments()) == 0 {
-			if n.Identity == "" {
-				res.SetError(errors.New("Identity not loaded!"), cmds.ErrNormal)
-				return
-			}
-			name = n.Identity.Pretty()
-
-		} else {
-			name = req.Arguments()[0]
+		name := req.Arguments()[0]
+		recursive, _, _ := req.Option("recursive").Bool()
+		depth := 1
+		if recursive {
+			depth = 0
 		}
 
-		output, err := n.Namesys.Resolve(n.Context(), name)
+		output, err := n.Namesys.Resolve(n.Context(), name, depth)
 		if err != nil {
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
 
-		// TODO: better errors (in the case of not finding the name, we get "failed to find any peer in table")
```

Comment:
On Thu, May 07, 2015 at 10:51:49PM -0700, Juan Batiz-Benet wrote:
> > -		// TODO: better errors (in the case of not finding the name, we get "failed to find any peer in table")
> 
> do we return better errors now?

Without a daemon:

  $ ipfs /ipfs/QmNonExistentKey
  panic?
  $ ipfs resolve /ipns/QmNonExistentKey
  Error: datastore: key not found
  $ ipfs resolve /dns/abc
  Error: not a valid domain name
  $ ipfs resolve /dns/example.com
  Error: could not resolve name.

With a daemon running, the DNS errors are the same, and the IPFS/IPNS
errors are:

  $ ipfs /ipfs/QmNonExistentKey
  Error: context deadline exceeded
  $ ipfs resolve /ipns/QmNonExistentKey
  Error: routing: not found

What sort of better errors were you looking for?